zetaclient/orchestrator/orchestrator.go

```diff
@@ -164,47 +164,13 @@ func (oc *Orchestrator) resolveSigner(app *zctx.AppContext, chainID int64) (inte
 		params := chain.Params()
 
 		// update zeta connector, ERC20 custody, and gateway addresses
-		zetaConnectorAddress := ethcommon.HexToAddress(params.GetConnectorContractAddress())
-		if zetaConnectorAddress != signer.GetZetaConnectorAddress() {
-			signer.SetZetaConnectorAddress(zetaConnectorAddress)
-			oc.logger.Info().
-				Str("signer.connector_address", zetaConnectorAddress.String()).
-				Msgf("updated zeta connector address for chain %d", chainID)
-		}
-		erc20CustodyAddress := ethcommon.HexToAddress(params.GetErc20CustodyContractAddress())
-		if erc20CustodyAddress != signer.GetERC20CustodyAddress() {
-			signer.SetERC20CustodyAddress(erc20CustodyAddress)
-			oc.logger.Info().
-				Str("signer.erc20_custody", erc20CustodyAddress.String()).
-				Msgf("updated erc20 custody address for chain %d", chainID)
-		}
-		if params.GatewayAddress != signer.GetGatewayAddress() {
-			signer.SetGatewayAddress(params.GatewayAddress)
-			oc.logger.Info().
-				Str("signer.gateway_address", params.GatewayAddress).
-				Msgf("updated gateway address for chain %d", chainID)
-		}
-
+		signer.SetZetaConnectorAddress(eth.HexToAddress(params.ConnectorContractAddress))
+		signer.SetERC20CustodyAddress(eth.HexToAddress(params.Erc20CustodyContractAddress))
+		signer.SetGatewayAddress(params.GatewayAddress)
```

Comment:
nit: it's out of this PR scope, but I'm curious regarding all this SetXXX functions that aims to set some value, but doesn't return an error in case of failure and just do `return`.

What would happen if for example `SetGatewayAddress` fails to set the Solana GW address? I suspect this could be failing into undefined behavior, and probably in the future we should consider returning an error whenever a SetXXX function fails.

Reply:
Linked https://github.com/zeta-chain/node/issues/2977#issuecomment-2470355952